File: plugins/dotnet/robocode.dotnet.control.api/src/implementation/robocode/control/events/BattleMessageEvent.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
﻿

using System;
using System.Collections.Generic;
using System.Text;

namespace Robocode.Control.Events
{
    /// <summary>
    /// Contains information about a <see cref="IRobocodeEngine.BattleMessage"/> event that is
    /// triggered when an informal message is sent from the game in the during the battle.
    /// </summary>
    /// <seealso cref="BattleMessageError"/>
    public class BattleMessageEvent : BattleEvent
    {
        private readonly string message;

        // Called by the game to create an instance of this event.
        // Note: This constructor should not be available in the API.
        internal BattleMessageEvent(string message)
            : base()
        {
            this.message = message;
        }

        /// <summary>
        /// Contains the informal message.
        /// </summary>
        /// <value>
        /// The informal message that was sent from the game during the battle.
        /// </value>
        public string Message
        {
            get { return message; }
        }
    }
}

File: robocode.api/src/main/java/robocode/control/events/BattleMessageEvent.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package robocode.control.events;


/**
 * A BattleMessageEvent is sent to {@link IBattleListener#onBattleMessage(BattleMessageEvent)
 * onBattleMessage()} when an informal message is sent from the game in the during the battle. 
 *
 * @see IBattleListener
 * @see BattleErrorEvent
 * 
 * @author Pavel Savara (original)
 * @author Flemming N. Larsen (contributor)
 *
 * @since 1.6.2
 */
public class BattleMessageEvent extends BattleEvent {
	private final String message;

	/**
	 * Called by the game to create a new BattleMessageEvent.
	 * Please don't use this constructor as it might change.
	 *
	 * @param message the informal message from the game.
	 */
	public BattleMessageEvent(String message) {
		super();
		this.message = message;
	}

	/**
	 * Returns the informal message.
	 *
	 * @return the informal message that was sent from the game during the battle.
	 */
	public String getMessage() {
		return message;
	}
}
